contracts/RiskSteward/RiskStewardReceiver.sol

```diff
@@ -276,101 +467,552 @@ contract RiskStewardReceiver is OApp, RiskStewardReceiverBase {
+                        baseOneIndex++;
+                        validProposalCount++;
+                    }
+                    writeUint256Cache(COUNT_CACHE_SLOT, destChainId, actionCount + 1);
+                } else {
+                    validProposalCount++;
+                }
+                updates[i] = update;
+            } else {
+                emit RiskParameterUpdateFailed(update.updateId, error);
+            }
+        }
+        return (baseOneIndex - 1, validProposalCount, updates);
+    }
+
+    /**
+     * @dev Silently validates a risk parameter update without reverting
+     * Checks that the update:
+     * - is the latest for the market and parameter,
+     * - is not expired,
+     * - is unprocessed,
+     * - has an active configuration,
+     * - has a valid destination (for non-cap updates),
+     * - has no active governance proposal,
+     * - and meets allowed range and debounce rules for local updates
+     *
+     * @param update The risk parameter update to validate.
+     * @return error The UPDATE_STATUS code indicating why the update is invalid, or NONE (0) if valid
+     * @return destChainId The destination chain ID extracted from the update's additional data
+     */
 
-        RiskParameterUpdate memory latestForMarketAndType = RISK_ORACLE.getLatestUpdateByParameterAndMarket(
+    function _validateUpdateStatus(
+        RiskParameterUpdate memory update
+    ) internal view returns (UPDATE_STATUS error, uint32 destChainId) {
+        (, destChainId) = _decodeAdditionalData(update.additionalData);
+
+        RiskParameterUpdate memory latestUpdate = RISK_ORACLE.getLatestUpdateByParameterAndMarket(
             update.updateType,
             update.market
         );
 
-        (, uint32 destChainId) = config.riskSteward.decodeAdditionalData(update.additionalData);
-
-        if (latestForMarketAndType.updateId != update.updateId) {
+        if (latestUpdate.updateId != update.updateId || update.timestamp + UPDATE_EXPIRATION_TIME < block.timestamp) {
             return (UPDATE_STATUS.EXPIRED, destChainId);
         }
 
+        UPDATE_STATUS processedStatus = processedUpdates[update.updateId];
+        if (
+            processedStatus == UPDATE_STATUS.PROCESSED ||
+            processedStatus == UPDATE_STATUS.SEND_TO_DESTINATION_CHAIN ||
+            processedStatus == UPDATE_STATUS.PROPOSED
+        ) {
+            return (processedStatus, destChainId);
+        }
+
+        RiskParamConfig memory config = riskParameterConfigs[update.updateType];
         if (!config.active) {
             return (UPDATE_STATUS.CONFIG_NOT_ACTIVE, destChainId);
         }
 
-        if (update.timestamp + UPDATE_EXPIRATION_TIME < block.timestamp) {
-            return (UPDATE_STATUS.EXPIRED, destChainId);
+        bool isLocalUpdate = destChainId == LAYER_ZERO_CHAIN_ID;
+        if (!isSupplyOrBorrowCapUpdate(update.updateType)) {
```

Comment:
Instead of having a function `isSupplyOrBorrowCapUpdate` to identify non governance updates we can add a flag to `RiskParamConfig`. This way if we introduce new types of RiskSteward contracts that doesn't need governance then we can configure them directly instead of upgrading the `RisStewardReceiver` contract.